sources/slack/__init__.py

```diff
@@ -161,4 +234,36 @@ def logs_resource() -> Iterable[TDataItem]:
         ):
             yield page_data
 
-    yield logs_resource
+    yield from (channels_resource, users_resource, logs_resource)
+
+    if table_per_channel:
+        for channel in fetched_selected_channels:
+            channel_name = channel["name"]
+            table_name = partial(table_name_func, channel_name)
+            messages_channel = dlt.resource(
+                per_table_messages_resource,
+                name=channel_name,
+                table_name=table_name,
+                primary_key=("channel", "ts"),
+                write_disposition="append",
+                columns={"blocks": {"data_type": "complex"}},
+            )(channel)
+
+            yield messages_channel
+            if replies:
+                yield messages_channel | dlt.transformer(
+                    get_thread_replies,
+                    name=channel_name + "_replies",
+                    table_name=partial(table_name_func, channel_name + "_replies"),
```

Comment:
`table_name_func` needs `type` and/or `subtype` to be present. maybe you should add them to the `context` of get_pages?

Reply:
These properties are already present in `page_data`.  

For example, so far I've seen only `thread_broadcast` subtype in replies and current source generates separate table `channel_name_replies_thread_broadcast`.

I've added `get("type", "")` though, in case there is some unexpected data.

<img width="756" alt="Screenshot 2024-02-19 at 12 37 18" src="https://github.com/dlt-hub/verified-sources/assets/9139429/cd1f8d3b-34a9-44d5-bf9f-761597f63e39">